src/connector/src/parser/json_parser.rs

```diff
@@ -20,19 +20,33 @@ use simd_json::{BorrowedValue, ValueAccess};
 use crate::impl_common_parser_logic;
 use crate::parser::common::simd_json_parse_value;
 use crate::parser::{SourceStreamChunkRowWriter, WriteGuard};
-use crate::source::SourceColumnDesc;
+use crate::source::{ErrorReportingContext, SourceColumnDesc};
 
 impl_common_parser_logic!(JsonParser);
 
 /// Parser for JSON format
 #[derive(Debug)]
 pub struct JsonParser {
     rw_columns: Vec<SourceColumnDesc>,
+    error_ctx: ErrorReportingContext,
```

Comment:
It is not actually used in `JsonParser`?

Reply:
Same problem for all the other Parsers.

Reply:
It is by `impl_common_parser_logic!`